### DISCUSSION TITLE
Pythonic ConfigurableRessources pydantic validation error

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18450] [Q&A]
Utilizing the Pythonic ConfigurableRessources and writing tests, i might have hit a wall of sorts,

I have a Resource that depends on another:

```python
class DatabricksIOManager(ConfigurableIOManager):
    client: UnityCatalogClient
```
Which i wanted to test by using the `mock_resource`
```python
manager = DatabricksIOManager(client=ResourceDefinition.mock_resource())
```
however i get pydantic validation errors
```python
Input should be a valid dictionary or instance of UnityCatalogClient [type=model_type, input_value=&lt;dagster._core.definition...bject at 0x7f2b437d3250&gt;, input_type=ResourceDefinition]
    For further information visit <https://errors.pydantic.dev/2.5/v/model_type>
```

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1701435462734909?thread_ts=1701435462.734909&cid=C01U954MEER).*

**Comment by C0DK** [comment]
Just to fix the markdown:

Utilizing the Pythonic ConfigurableRessources and writing tests, i might have hit a wall of sorts,

I have a Resource that depends on another:

```python
class DatabricksIOManager(ConfigurableIOManager):
    client: UnityCatalogClient
```

Which i wanted to test by using the mock_resource

```python
    manager = DatabricksIOManager(client=ResourceDefinition.mock_resource())
```


however i get pydantic validation errors
```
Input should be a valid dictionary or instance of UnityCatalogClient [type=model_type, input_value=<dagster._core.definition...bject at 0x7f2b437d3250>, input_type=ResourceDefinition]
    For further information visit https://errors.pydantic.dev/2.5/v/model_type
```

**Reply by smackesey** [reply]
@benpankow do we have a version of `@mock_resource` that works with new-style resources?
